middleware/introspection/latency.go

```diff
@@ -0,0 +1,63 @@
+package introspection
+
+import (
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/unit"
+)
+
+// pathKey is the label for the http path a latency
+// metric pertains to.
+var pathKey = key.New("http.path")
+var methodKey = key.New("http.method")
+
+// Handler wraps the provided http.Handler and provides
+// latency recordings per incoming http path using the
+// open telemetry library
+//
+// The passed in pattern will be used as a metric key along
+// with the http request method.
+// For example a resulting prometheus latency metric will take the form
+// "clair_http_latence{http_path={path}, http.method={method}}"
+func Handler(next http.Handler, path string) http.Handler {
+	meter := global.MeterProvider().Meter("clair")
+	late := meter.NewInt64Measure("clair.http.latency",
+		metric.WithDescription("latency of http requests"),
+		metric.WithUnit(unit.Milliseconds),
+		metric.WithAbsolute(true),
+		metric.WithKeys(pathKey),
+		metric.WithKeys(methodKey),
```

Comment:
@hdonnay this api is whack, why must I put my keys here if I also provide them to h.latency.Record.

Reply:
You set the keys here and record the values at Record time. You can use the Bind method to bind the labels early, but can't in this case because they're not known beforehand.